### PR TITLE
Remove `@slow` for `test_eager_matches_sdpa_inference`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

And make it less flaky

- Use smaller (short) inputs
- Use larger epsilon in norm layers

### CODE REVIEW

File: tests/generation/test_utils.py [hunk 1]
```diff
@@ -185,16 +185,10 @@ def _get_logits_processor_kwargs(self, do_sample=False, config=None):
...
-            )
-            video_token_index = config.video_token_index if hasattr(config, "video_token_index") else None
-            if image_token_index is not None and image_token_index < config.get_text_config().vocab_size:
-                logits_processor_kwargs["bad_words_ids"].append([image_token_index])
-            if video_token_index is not None and video_token_index < config.get_text_config().vocab_size:
-                logits_processor_kwargs["bad_words_ids"].append([video_token_index])
+            for key in ["image_token_index", "image_token_id", "video_token_index", "video_token_id", "vision_start_token_id"]:
+                token_index = getattr(config, key, None)
+                if token_index is not None and token_index < config.get_text_config().vocab_size:
+                    logits_processor_kwargs["bad_words_ids"].append([token_index])
```
Comment (ydshieh) [inline tests/generation/test_utils.py:191]: make it more general.

`vision_start_token_id` is required for `qwen2_vl`

File: tests/models/albert/test_modeling_albert.py [hunk 2]
```diff
@@ -288,6 +289,12 @@ def setUp(self):
         self.model_tester = AlbertModelTester(self)
         self.config_tester = ConfigTester(self, config_class=AlbertConfig, hidden_size=37)
 
+    @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])
+    @require_torch_sdpa
+    @unittest.skip("Albert requires `head_mask` which is currently not done in this test.")
+    def test_eager_matches_sdpa_inference(self):
+        pass
```
Comment (gante) [inline tests/models/albert/test_modeling_albert.py:296]: On skips like this, on Albert and other models: the test pulls the main input and the attention mask to manipulate them, finally sending them to the model. We could `pop` these items from `inputs_dict`, and then pass `**inputs_dict` to the model (e.g. `model_eager(**prepared_inputs, **inputs_dict)`) -- I think then we wouldn't need to skip tests due to missing inputs 🤗

File: tests/models/idefics/test_modeling_idefics.py [hunk 3]
```diff
@@ -353,6 +352,12 @@ def _prepare_for_class(self, inputs_dict, model_class, return_labels=False):
 
         return inputs_dict
 
+    @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])
+    @require_torch_sdpa
+    @unittest.skip("Idefics requires both text and image inputs which is currently not done in this test.")
```
Comment (gante) [inline tests/models/idefics/test_modeling_idefics.py:357]: (same comment as in Albert)

File: tests/models/mimi/test_modeling_mimi.py [hunk 4]
```diff
@@ -411,8 +411,10 @@ def test_identity_shortcut(self):
 
     @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])
     @require_torch_sdpa
-    @slow
     def test_eager_matches_sdpa_inference(self, torch_dtype: str):
```
Comment (gante) [inline tests/models/mimi/test_modeling_mimi.py:414]: Can we add a comment at the top explaining why we need to overwrite this test in mimi? 🤗 

If the test is overwritten only to skip the case `torch_dtype == "float16" and torch_device == "cpu"`, can we delete the overwritten test and skip only that case?

File: tests/models/musicgen/test_modeling_musicgen.py [hunk 5]
```diff
@@ -452,7 +452,6 @@ def test_flash_attn_2_inference_equivalence_right_padding(self):
 
     @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])
     @require_torch_sdpa
-    @slow
```
Comment (gante) [inline tests/models/musicgen/test_modeling_musicgen.py:455]: Can't we just delete the test? (it has `# Copied from tests.test_modeling_common.ModelTesterMixin.test_eager_matches_sdpa_inference` and it inherits the mixin, so it should run the original test!)

File: tests/models/musicgen_melody/test_modeling_musicgen_melody.py [hunk 6]
```diff
@@ -460,7 +460,6 @@ def test_flash_attn_2_inference_equivalence_right_padding(self):
 
     @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])
     @require_torch_sdpa
-    @slow
```
Comment (gante) [inline tests/models/musicgen_melody/test_modeling_musicgen_melody.py:463]: Same comment as in musicgen

**Review by gante — APPROVED** [review]
Yay fewer slow tests 🙌 

Added a few questions/suggestions to see if we can remove a few more overwritten cases 😈

**Comment by gante** [comment]
Tagging @zucchini-nlp to double-check VLM test changes :)

File: tests/models/musicgen/test_modeling_musicgen.py (hunk 5 above)
Comment (ydshieh) [reply tests/models/musicgen/test_modeling_musicgen.py:455]: no, it fails. I don't dive into why it's failing (input issues) though.

File: tests/models/albert/test_modeling_albert.py (hunk 2 above)
Comment (ydshieh) [reply tests/models/albert/test_modeling_albert.py:296]: Maybe let's merge and do this in a follow-up PR. It's never worked before.

File: tests/models/musicgen/test_modeling_musicgen.py (hunk 5 above)
Comment (gante) [reply tests/models/musicgen/test_modeling_musicgen.py:455]: Interesting -- in that case, how does `# Copied from ` work? 👀

File: tests/models/idefics/test_modeling_idefics.py [hunk 7]
```diff
@@ -316,7 +316,6 @@ def prepare_pixel_values(self):
         return floats_tensor([self.batch_size, self.num_channels, self.image_size, self.image_size])
 
     @require_torch_sdpa
-    @slow
```
Comment (zucchini-nlp) [inline tests/models/idefics/test_modeling_idefics.py:319]: i think this test dont need skip anymore since we dont check if model has SDPA layers within this test anymore. But it can be skipped due to the same flakiness

File: tests/models/llava_next_video/test_modeling_llava_next_video.py [hunk 8]
```diff
@@ -125,11 +125,12 @@ def __init__(
...
-        self.encoder_seq_length = 127
-        self.image_grid_pinpoints = [[32, 32]]
-        self.num_image_tokens = 88
-        self.num_video_tokens = 32
+
+        self.image_grid_pinpoints = [[16, 16]]
+        self.num_image_tokens = 24
+        self.num_video_tokens = 8
         self.seq_length = seq_length + self.num_image_tokens + self.num_video_tokens
+        self.encoder_seq_length = self.seq_length
```
Comment (zucchini-nlp) [inline tests/models/llava_next_video/test_modeling_llava_next_video.py:133]: not very related to this PR but lets clean up also, maybe we dont need `encoder-seq-length` in llavas anymore because it was a hack to account for image tokens previously

File: tests/models/qwen2_vl/test_modeling_qwen2_vl.py [hunk 9]
```diff
@@ -166,6 +166,8 @@ def prepare_config_and_inputs_for_common(self):
         input_ids = ids_tensor([self.batch_size, self.seq_length], self.vocab_size)
         attention_mask = torch.ones(input_ids.shape, dtype=torch.long, device=torch_device)
 
+        input_ids[:, -1] = self.pad_token_id
```
Comment (zucchini-nlp) [inline tests/models/qwen2_vl/test_modeling_qwen2_vl.py:169]: for my understanding, any reason why last token has to be a pad?

**Review by zucchini-nlp — APPROVED** [review]
Thanks, LGTM! Left a few questions and things we can clean up more

File: tests/test_modeling_common.py [hunk 10]
```diff
@@ -3987,12 +3990,29 @@ def get_mean_reldiff(failcase, x, ref, atol, rtol):
...
+            config.layer_norm_epsilon = 1.0
+
+            for attr in ["text_config", "vision_config", "text_encoder", "audio_encoder", "decoder"]:
+                if hasattr(config, attr):
+                    getattr(config, attr).rms_norm_eps = 1.0
+                    getattr(config, attr).layer_norm_eps = 1.0
+                    getattr(config, attr).norm_eps = 1.0
+                    getattr(config, attr).norm_epsilon = 1.0
+                    getattr(config, attr).layer_norm_epsilon = 1.0
+
```
Comment (ArthurZucker) [inline tests/test_modeling_common.py:4015]: cam you add a comment about why we do this?

**Review by ArthurZucker — APPROVED** [review]
thanks 🤗

File: tests/models/idefics/test_modeling_idefics.py (hunk 7 above)
Comment (ydshieh) [reply tests/models/idefics/test_modeling_idefics.py:319]: It still has input preparation issues, where I added below to another model test class

> "Idefics requires both text and image inputs which is currently not done in this test."

As mentioned in a reply to Joao's comment, let's try to do it in a follow up PR

File: tests/models/qwen2_vl/test_modeling_qwen2_vl.py (hunk 9 above)
Comment (ydshieh) [reply tests/models/qwen2_vl/test_modeling_qwen2_vl.py:169]: to avoid index error 

(modeling code)
> vision_tokens = input_ids[vision_start_indices + 1]
